struct.rkt

```diff
@@ -62,3 +90,53 @@
 (provide *Range
          make-Range
          Range->hash)
+
+(struct FormattingOptions
+  (tabSize
+   insertSpaces
+   trimTrailingWhitespace
+   insertFinalNewline
+   trimFinalNewlines
+   key))
+
+(define/contract (make-FormattingOptions #:tabSize tabSize
+                                         #:insertSpaces insertSpaces
+                                         #:trimTrailingWhitespace [trimTrailingWhitespace undef-object]
+                                         #:insertFinalNewline [insertFinalNewline undef-object]
+                                         #:trimFinalNewlines [trimFinalNewlines undef-object]
+                                         #:key [key undef-object])
+  (-> #:tabSize uinteger?
+      #:insertSpaces boolean?
+      #:trimTrailingWhitespace (undef/c boolean?)
+      #:insertFinalNewline (undef/c boolean?)
+      #:trimFinalNewlines (undef/c boolean?)
+      #:key (undef/c hash?)
+      FormattingOptions?)
+
+  (FormattingOptions tabSize
+                     insertSpaces
+                     trimTrailingWhitespace
+                     insertFinalNewline
+                     trimFinalNewlines
+                     key))
+
+(define (jsexpr->FormattingOptions js)
```

Comment:
`js` often stands for **JavaScript**, I think we better use abbreviation like `json` instead of compress it futher

Reply:
we can use name `jsexpr` instead, the same as the json library used.

Reply:
ok